addons/web/static/src/js/fields/basic_fields.js

```diff
@@ -508,6 +512,49 @@ var NumericField = InputField.extend({
         }
         return this._super(value, options);
     },
+
+    //--------------------------------------------------------------------------
+    // Handlers
+    //--------------------------------------------------------------------------
+
+    /**
+     * @private
+     * @param {KeyboardEvent} ev
+     */
+    async _onKeydown(ev) {
```

Comment:
I think you can use _onInput of debouced fields
that listen on `input` events
https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/input_event

Reply:
`input` only fires when the value of an `<input>` has already changed.
It could bring a visual issue when replacing the decimal separator, as the browser may already have rendered the « unchanged separator » .